hi-ml-histopathology/src/histopathology/datamodules/base_module.py

```diff
@@ -103,43 +101,80 @@ def get_splits(self) -> Tuple[TilesDataset, TilesDataset, TilesDataset]:
         """Create the training, validation, and test datasets"""
         raise NotImplementedError
 
-    def prepare_data(self) -> None:
-        if self.precache_location != CacheLocation.NONE:
-            self._load_dataset(self.train_dataset, stage='train', shuffle=True)
-            self._load_dataset(self.val_dataset, stage='val', shuffle=True)
-            self._load_dataset(self.test_dataset, stage='test', shuffle=True)
-
     def _dataset_pickle_path(self, stage: str) -> Optional[Path]:
         if self.cache_dir is None or self.cache_mode == CacheMode.NONE:
             return None
         return self.cache_dir / f"{stage}_dataset.pt"
 
+    def prepare_data(self) -> None:
+        pass
+
+    def _get_transformed_dataset(
```

Comment:
this method, that is the only one making use of the caching parameters, is currently called only in the TilesDataModule. I see two options:
* we support caching only for the TilesDataModule and we move this method inside that child class, with also all the respective parameters.
* we decide to keep the caching option also in the SlidesDataModule, we keep this method and caching parameters in the base_module and call _get_transformed_dataset inside the `load_dataset` implementation of `SlidesDataModule`.

For the sake of keeping this first slide implementation as simple as possible I would vote for the second

Reply:
Oh, I actually just moved all the caching options to TilesDatModule only. As you said last time, caching would be detrimental if we want to have random tiling.

Reply:
Sounds good to me. This method and cache parameters should then not be in the base class